versions/development/SPEC.md

```diff
@@ -2275,6 +2277,30 @@ The latter case is very likely an error case, and this `--val=` part should be l
 python script.py ${"--val=" + val}
 ```
 
+## The `None` argument
+
+Sometimes you need to define an output that is optional, because it depends on a command parameter. For example:
+```wdl
+task test
+  input {
+    File inputFile
+    String outputPath
+    Boolean logFile
+  }
+  command {
+    test ~{true="--log" false="" logFile} inputFile outputPath
+  }
+  output {
+    File? logFile = if logFile then "test.log" else None
+  }
+}
+```
+If `logFile` is true, the command will evaluate to `test --log inputFile outputPath` and a log will be created.
+If `logFile` is false, the command will evaluate to `test inputFile outputPath` and log will not be created.
+
+Using a non-optional `File` will crash when `logFile` is false. The `None` directive is there for these and
```

Comment:
> Using a non-optional `File` will crash when `logFile` is false.

Do you mean that `File logFile = if logFile then "test.log" else None` is not valid WDL and will result in a compiler error? That seems correct to me, as the WDL compiler will determine that the type of the `if` expression is `File?` and then disallow the assignment.

> The `None` directive is there for these and other cases where an optional variable is conditionally defined.

Am not sure what you mean by `directive` in this context. I think it would be more general to define `None` as a literal that can be used anywhere an expression with an optional type is valid. E.g. it's natural to think of
```
input {
    File? inputFile
}
```
being shorthand for
```
input {
    File? inputFIle = None
}
```

Reply:
FWIW I also prefer the term `literal` over `directive`